### ISSUE TITLE
beta.7 - match.params returns empty object

### ISSUE THREAD

**Issue by FourwingsY** [ReactTraining/react-router#4646]
## Version
4.0.0-beta.7

## Steps to reproduce
```
<Route path="/:menu" render={() => <Component />} />

// go to path: /a/b/c
// in Component.render()
console.log(this.props.match.params)
```
```
// notice: this behavior only shown when withRouter is applied to Component 
class Component extends React.Component {
   ...
}

export default withRouter(Component)
```

## Expected Behavior
expected to be `{menu: "a"}`
and [Route Tester](https://pshrmn.github.io/route-tester/#/a/b/c) does, also beta.6 does.

## Actual Behavior
`{}`

**Comment by Shaxmatist** [comment]
You should to remove `withRouter` wrapper. It helps.

**Comment by pshrmn** [comment]
You do not have to wrap a component rendered by a `<Route>` in `withRouter` because it is already being provided the same props (`history`, `match`, and `location`).

That said, you are running into a bug, but it is fixed in #4643.
